### DISCUSSION TITLE
Examples

### DISCUSSION THREAD

**Discussion by alex-hedley** [alex-hedley/tdd-example#2] [Ideas]
What examples should we create?

**Comment by alex-hedley** [comment]
Tests from the start with a new method

**Comment by alex-hedley** [comment]
Legacy method with no tests - add some

**Comment by alex-hedley** [comment]
Legacy method with tests but also bug - add test to fix bug

**Comment by alex-hedley** [comment]
Specification Testing - [Spek](https://www.spekframework.org)

**Comment by mpbostock** [comment]
Here's a few ideas I had so far:

- The repo could have different branches (it may even be empty when on main) Or do we have multiple repos?
- Each branch practises a different aspect
- One branch could be a simple Kata. On that branch every red/green/refactor step is committed one at a time. People could branch off from a specific commit and carry on from there to practise maybe different stages of the solution
- One branch could have some "legacy" code that has a few bugs in and the idea is to reproduce the bugs with failing tests and then fix the bug to make the test pass
- One branch could have the gilded rose or similar refactoring kata. This one will focus on writing tests for the functionality but then refactor while keeping the tests passing

Any other thoughts/ideas?
